force-app/main/default/classes/Callable_API_TEST.cls

```diff
@@ -117,6 +119,53 @@ private with sharing class Callable_API_TEST {
         Test.stopTest();
     }
 
+    @IsTest
+    private static void shouldCallEnableAdvancedMappingOnAction() {
+        Callable_API callableApi = new Callable_API();
+        AdvancedMappingEnablementServiceMock mappingServiceMock = new AdvancedMappingEnablementServiceMock();
+        callableApi.advancedMappingEnablementService = stubFor(mappingServiceMock);
+
+        Test.startTest();
+            callableApi.call(ADVANCED_MAPPING_ACTION, null);
+        Test.stopTest();
+
+        System.assert(mappingServiceMock.enableAdvancedMappingCalled, 'The enable advanced mapping service method ' +
+                'should be called.');
+    }
+
+    @IsTest
+    private static void shouldNotCallEnableAdvancedMappingIfMappingIsEnabled() {
+        Data_Import_Settings__c dataImportSettings = new Data_Import_Settings__c();
+        dataImportSettings.Field_Mapping_Method__c = BDI_MigrationMappingUtility.DATA_IMPORT_FIELD_MAPPING;
+        UTIL_CustomSettingsFacade.setDataImportSettings(dataImportSettings);
+
+        Callable_API callableApi = new Callable_API();
+        AdvancedMappingEnablementServiceMock mappingServiceMock = new
+                AdvancedMappingEnablementServiceMock();
+        callableApi.advancedMappingEnablementService = stubFor(mappingServiceMock);
+
+        Test.startTest();
+            callableApi.call(ADVANCED_MAPPING_ACTION, null);
+        Test.stopTest();
+
+        System.assert(!mappingServiceMock.enableAdvancedMappingCalled, 'The enable advanced mapping service method ' +
+                'should not be called.');
+    }
+
+    @IsTest
+    private static void shouldCallEnableGiftEntryOnAction() {
```

Comment:
I see you got the already enabled scenario covered for Advanced Mapping, but it'd be nice to have it for Gift Entry too. Great test coverage otherwise!

Reply:
added!